### ISSUE TITLE
Error on load

### ISSUE THREAD

**Issue by Hardeh** [AdalynBlack/Loadstone#1]
```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
MonoMod.Utils.ReflectionHelper.GetRuntimeHashedFullName (Mono.Cecil.AssemblyNameReference asm) (at <6733e342b5b549bba815373898724469>:IL_0017)
MonoMod.Utils.ReflectionHelper._ResolveReflection (Mono.Cecil.MemberReference mref, System.Reflection.Module[] modules) (at <6733e342b5b549bba815373898724469>:IL_00AF)
MonoMod.Utils.ReflectionHelper.ResolveReflection (Mono.Cecil.TypeReference mref) (at <6733e342b5b549bba815373898724469>:IL_0000)
MonoMod.Utils._DMDEmit+<>c__DisplayClass5_0.<Generate>b__0 (Mono.Cecil.Cil.VariableDefinition var) (at <6733e342b5b549bba815373898724469>:IL_000C)
System.Linq.Enumerable+SelectIListIterator`2[TSource,TResult].ToArray () (at <dab7f68612224ba3ae40f651d44f9d4c>:IL_0034)
System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) (at <dab7f68612224ba3ae40f651d44f9d4c>:IL_001F)
MonoMod.Utils._DMDEmit.Generate (MonoMod.Utils.DynamicMethodDefinition dmd, System.Reflection.MethodBase _mb, System.Reflection.Emit.ILGenerator il) (at <6733e342b5b549bba815373898724469>:IL_015C)
MonoMod.Utils.DMDEmitDynamicMethodGenerator._Generate (MonoMod.Utils.DynamicMethodDefinition dmd, System.Object context) (at <6733e342b5b549bba815373898724469>:IL_0332)
MonoMod.Utils.DMDGenerator`1[TSelf].Generate (MonoMod.Utils.DynamicMethodDefinition dmd, System.Object context) (at <6733e342b5b549bba815373898724469>:IL_0013)
MonoMod.Utils.DynamicMethodDefinition.Generate (System.Object context) (at <6733e342b5b549bba815373898724469>:IL_013A)
MonoMod.Utils.DynamicMethodDefinition.Generate () (at <6733e342b5b549bba815373898724469>:IL_0000)
(wrapper dynamic-method) MonoMod.RuntimeDetour.ILHook+Context.DMD<MonoMod.RuntimeDetour.ILHook+Context::Refresh>(MonoMod.RuntimeDetour.ILHook/Context)
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition.Trampoline<MonoMod.RuntimeDetour.ILHook+Context::Refresh>?-693053586(object)
HarmonyLib.Internal.RuntimeFixes.StackTraceFixes.OnILChainRefresh (System.Object self) (at <474744d65d8e460fa08cd5fd82b5d65f>:IL_0000)
MonoMod.RuntimeDetour.ILHook.Apply () (at <4e2760c7517c4ea79c633d67e84b319f>:IL_0059)
HarmonyLib.PatchFunctions.ReversePatch (HarmonyLib.HarmonyMethod standin, System.Reflection.MethodBase original, System.Reflection.MethodInfo postTranspiler, System.Reflection.MethodInfo postManipulator) (at <474744d65d8e460fa08cd5fd82b5d65f>:IL_015D)
Rethrow as HarmonyException: IL Compile Error (unknown location)
HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) (at <474744d65d8e460fa08cd5fd82b5d65f>:IL_0045)
HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:IL_0095)
HarmonyLib.Harmony.PatchAll (System.Type type) (at <474744d65d8e460fa08cd5fd82b5d65f>:IL_0008)
HarmonyLib.Harmony.CreateAndPatchAll (System.Type type, System.String harmonyInstanceId) (at <474744d65d8e460fa08cd5fd82b5d65f>:IL_001E)
Loadstone.Loadstone.Awake () (at <445e04fe67d044aca995cc82fa9519d6>:IL_0091)
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Rendering.HighDefinition.HDRenderPipelineAsset:OnEnable()
```

Mod list:
```
BepInEx-BepInExPack-5.4.2100
Hardy-LCMaxSoundsFix-1.1.0
Evaisa-HookGenPatcher-0.0.5
Evaisa-LethalLib-0.14.2
zealsprince-Locker-0.13.1
IntegrityChaos-GraphicsAPI-1.0.0
IntegrityChaos-LCCutscene-1.0.0
IntegrityChaos-Diversity-2.0.3
x753-Peepers-0.9.6
jaspercreations-Scopophobia-1.1.1
NomnomAB-RollingGiant-2.5.2
x753-Mimics-2.4.1
Ozone-Runtime_Netcode_Patcher-0.2.5
readthisifbad-SnatchinBracken-1.4.3
SillySquad-Arachnophilia-1.6.1
Rolevote-The_Fiend-1.0.5
Ccode_lang-SirenHead-1.0.1
SolidStone-ShockwaveDroneEnemy-0.3.0
ShaosilGaming-GeneralImprovements-1.1.11
ShaosilGaming-FlashlightFix-1.1.2
Lordfirespeed-OdinSerializer-2022.11.9
IAmBatby-LethalLevelLoader-1.1.6
xilophor-LethalNetworkAPI-2.1.6
zealsprince-Malfunctions-1.8.2
FlipMods-ReservedItemSlotCore-1.8.17
FlipMods-ReservedFlashlightSlot-1.6.3
FlipMods-ReservedWalkieSlot-1.6.2
Piggy-LC_Office-1.1.5
skidz-PoolRooms-0.1.18
Dantor-Dantors_Mental_Hospital-1.1.1
Alice-ScarletDevilMansion-1.3.2
Badham_Mods-SCPFoundationDungeon-3.0.0
Major_And_Skiz-MoreInteriors-2.5.0
scoopy-Scoopys_Variety_Mod-1.2.0
mrgrm7-LethalCasino-0.4.2
Monkeytype-HideChat-1.0.0
Zduniusz-LBtoKG-1.0.0
x753-More_Suits-1.4.1
qwbarch-BarchLib-1.0.0
Bobbie-NAudio-2.2.2
Bobbie-UniTask-2.5.0
qwbarch-Mirage-1.1.0
xCeezy-LethalEscape-0.8.5
Swaggies-BetterEXP-1.2.2
dopadream-Some_Buyable_Suits_Fix-1.0.3
malco-Lategame_Upgrades-3.3.0
notnotnotswipez-MoreCompany-1.8.1
HGG-JigglePhysicsPlugin-1.1.2
Mhz-MoreHead-1.2.6
LethalAPI-LethalAPI_Terminal-1.0.1
mattymatty-LobbyControl-2.2.6
Rune580-LethalCompany_InputUtils-0.6.3
KlutzyBubbles-BetterEmotes-1.4.5
PotatoCoders-MinimumQuotaFinder-1.1.3
Suskitech-AlwaysHearActiveWalkies-1.4.4
TanmanG-Forest_Giant_Motionsense-1.0.2
Woecust-Immersive_Visor-0.2.1
mrov-WeatherTweaks-0.4.6
Fredolx-MeteoMultiplier-1.1.3
Sigurd-CSync-3.1.1
Evaisa-LethalThings-0.9.4
Scintesto-Monster_Plushies-1.1.2
bigmcnugget-VanillaContentExpansion-0.1.8
Refract-LethalLevelLoaderFixed-1.1.0
Zaggy1024-PathfindingLagFix-1.2.1
NutNutty-SellTracker-1.2.1
veri-ShipWindows-1.3.6
CatsArmy-SpaceSunShine-1.2.1
KawaiiBone-Remnants-1.1.7
mrov-TerminalFormatter-0.0.8
AdiBTW-Loadstone-0.0.1
```

**Comment by AdalynBlack** [comment]
I'm currently looking into this issue, and it appears that the way the development profile I set up works is behaving differently from a fresh profile. Even with no other mods installed, the same error is happening. I'll try to get a fix out as soon as possible

**Comment by AdalynBlack** [comment]
I've figured out why my dev environment works differently, and you might be able to use the same change as a temporary fix until I can resolve the issue fully. By changing BepInEx.cfg's HarmonyBackend option from `auto` to `cecil`, it should work. I'm currently try to make it work on the default backend, but it could take a bit

**Comment by Slydylan** [comment]
well i set it to cecil and i am now watching advanced company do something called 'added DMDASM.XXXXXXX.Cecil' thousands and thousands of times and it looks like ill never reach the load screen

**Comment by Slydylan** [comment]
but it did do something to the load times even without setting it to cecil..

**Comment by AdalynBlack** [comment]
interesting
I'm guessing that that particular mod just doesn't work with the cecil backend
I'll keep poking around and see if I can figure it out on my end. Also, while level loading might work for the most part, the particular function that's erroring is responsible for spawning in the doors between rooms, so there will likely not be any doors spawning between rooms until I can find a fix

**Comment by Slydylan** [comment]
looks like something is defaulting it to generate properly as i've not noticed anything as of yet

**Comment by AdalynBlack** [comment]
perhaps that section of patches isn't being applied? Or maybe Lethal Company handles doors separately from the map generator, although I don't think that it does if my memory serves correctly

**Comment by AdalynBlack** [comment]
Ok! I think I've finally located the root of the issue. It seems to be a potential bug in MonoMod, which lies underneath of BepInEx, and was causing files to not be loaded when they should be. I've created a workaround that seems to resolve the error. I'm uploading the new version the thunderstore now, and the update should go live within 30 minutes to an hour (thunderstore takes a while to actually display updates for some reason)
